vary error for callback vs promise

bruce-lindsay opened the pull request:
https://github.com/cucumber/cucumber-js/issues/1019

charlierudolph commented on features/step_definition_timeouts.feature:
```diff
@@ -81,7 +81,7 @@ Feature: Step definition timeouts
     And the output contains the text:
       """
-      function timed out after 500 milliseconds
+      function timed out
```
Can we do what was done earlier in this file?

charlierudolph commented on features/before_after_all_hook_timeouts.feature:
```diff
@@ -29,7 +29,8 @@ Feature: before / after all hook timeouts
       | a handler errored, process exiting        |
-      | function timed out after 500 milliseconds |
+      | function timed out                        |
+      | 500 milliseconds                          |
```
Can we update this to: `function timed out, ensure the callback is executed within 500 milliseconds`?

bruce-lindsay commented:
@charlierudolph sure thing. Glad to be able to help. I addressed your feedback in the latest commit. I thought I was being smart by reducing some of the repetitiveness but this is more consistent.  Let me know if I missed anything else.

aslakhellesoy commented:
Hi @bruce-lindsay,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber
